src/scripting/lua_game.cc

```diff
@@ -690,8 +693,11 @@ int LuaPlayer::hide_fields(lua_State* L) {
 	Widelands::Player& p = get(L, game);
 
 	luaL_checktype(L, 2, LUA_TTABLE);
-	const Widelands::HideOrRevealFieldMode mode = (!lua_isnone(L, 3) && luaL_checkboolean(L, 3)) ?
+	const std::string state = luaL_checkstring(L, 3);
+	const Widelands::HideOrRevealFieldMode mode = (!lua_isnone(L, 3) && state == "permanent") ?
```

Comment:
Shouldn't something like this do work? (Sorry for the python-like syntax)

```	
        if state == "permanent" then 
              mode = Widelands::HideOrRevealFieldMode::kHide;
        elseif state == "seen" then 
              mode = Widelands::HideOrRevealFieldMode::kUnreveal;
        else 
             mode = Widelands::HideOrRevealFieldMode::kUnexplore;
```

Those nested ternary operators are difficult to understand for me...